src/index.js

```diff
@@ -39,7 +40,7 @@ class ErrorBoundary extends React.Component {
         return fallback
       } else if (typeof fallbackRender === 'function') {
         return fallbackRender(props)
-      } else if (typeof FallbackComponent === 'function') {
+      } else if (isValidElementType(FallbackComponent)) {
```

Comment:
What if instead we just do this:

```suggestion
      } else if (FallbackComponent) {
```

I don't want to worry about the extra dep and React will do a good job telling people that something's wrong when we try to call React.createElement with it.